crates/duty-tracker/src/contract_state_machine.rs

```diff
@@ -651,6 +681,13 @@ impl ContractCfg {
     pub fn tx_build_context(&self) -> TxBuildContext {
         self.operator_table.tx_build_context(self.network)
     }
+
+    /// Returns the transaction ID of the deposit request for this contract.
+    pub fn deposit_request_txid(&self) -> Txid {
+        self.deposit_tx.psbt().unsigned_tx.input[0]
+            .previous_output
+            .txid
+    }
```

Comment:
rewrite the state machine one against this definition.

Reply:
Done in b97827c2ec400a0af41d9092bb91e5fc2c6626cf

Reply:
No I mean literally do `self.cfg().deposit_request_txid()`